sdk/core/azure-core/azure/core/tracing/common.py

```diff
@@ -0,0 +1,79 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+from os import environ
+import re
+
+from azure.core.tracing.context import tracing_context
+from azure.core.tracing.abstract_span import AbstractSpan
+from azure.core.settings import settings, get_opencensus_wrapper
+
+
+def set_span_contexts(wrapped_span, span_instance=None):
+    # type: (AbstractSpan, AbstractSpan) -> None
+    tracing_context.current_span.set(wrapped_span)
+    impl_wrapper = settings.tracing_implementation()
+    tracing_context.tracing_impl.set(impl_wrapper.__class__)
+    if wrapped_span is not None:
+        span_instance = wrapped_span.span_instance
+    if impl_wrapper is not None:
+        impl_wrapper.set_current_span(span_instance)
+
+
+def get_parent(*args, **kwargs):
+    # type: (Any) -> Tuple(Any, Any, Any)
+    """Returns the parent span that of the span that represents the function and the spans before that parent span"""
```

Comment:
I'm confused by this function. The name suggests it returns one span. The docstring is tough to parse and kind of supports that (a serial comma would help clarify). However, the type hint promises an arbitrary triple, and the function secretly accepts a `parent_span` argument. If I already have `parent_span`, do I call `get_parent` to get its parent? What's the triple for?

Reply:
renamed to `get_parent_and_original_contexts`

Reply:
I fixed it to get_parent_span